packages/api-queries/src/plugin.ts

```diff
@@ -22,3 +24,18 @@ export const wpOrgPluginQuery = ( slug: string, locale: string ) =>
 		queryKey: [ 'wp-org-plugin', slug, locale ],
 		queryFn: () => fetchWpOrgPlugin( slug, locale ),
 	} );
+
+export const installPluginMutation = () =>
+	mutationOptions( {
+		mutationFn: ( vars: { siteId: number | string; slug: string } ) =>
+			installPlugin( vars.siteId, vars.slug ),
+		onSuccess: ( _data, vars ) => {
+			// Refresh the plugin data so usePlugin sees the newly installed plugin.
+			// We invalidate both the per-site plugins list and the aggregated plugins
+			// query that `usePlugin` relies on to compute sitesWith/WithoutThisPlugin.
+			if ( typeof vars.siteId === 'number' ) {
```

Comment:
`siteId` should always be number I think 🤔 in what case is it not?

Reply:
Nice one! I've updated the type number.